pkg/controller/controller_instance.go

```diff
@@ -341,6 +370,126 @@ func (c *controller) initOrphanMitigationCondition(instance *v1beta1.ServiceInst
 	return false, nil
 }
 
+// setRetryBackoffRequired marks the specified instance/generation as needing a
+// delay before the next provision/update is attempted.  We always set this flag
+// before attempting a provision or update operation in case we must retry.  This
+// will eventually be cleared by the background worker running
+// purgeExpiredRetryEntries() or when the operation is successful.
+func (c *controller) setRetryBackoffRequired(instance *v1beta1.ServiceInstance) {
+	pcb := pretty.NewInstanceContextBuilder(instance)
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(instance)
+	if err != nil {
+		glog.Errorf(pcb.Messagef("Couldn't create a key for object %+v: %v", instance, err))
+		return
+	}
+
+	c.instanceOperationRetryQueue.mutex.Lock()
+	defer c.instanceOperationRetryQueue.mutex.Unlock()
+	retryEntry, found := c.instanceOperationRetryQueue.instances[key]
+	if !found || retryEntry.generation != instance.Generation {
```

Comment:
@jboyd01 I think we may need to also store and compare instance's `UID`.
@joshk has reported that when he deletes an old instance and creates a new one under the same name, he gets a `RetryBackoff` event immediately. I suspect this could happen when the old instance had a `Generation = 1` (created and never change) - and the new instance will get the same.

Reply:
Interesting.   Thanks for digging in on it @nilebox.